### ISSUE TITLE
Cannot open encrypted G Suite Team Drive

### ISSUE THREAD

**Issue by jpdribbler** [kaczmarkiewiczp/rcloneExplorer#85]
Hi!

First of all thanks a lot for this amazing app! 

Unfortunately i cannot open my encrypted TeamDrive in your app, it looks like something with my config is not correct. On linux and windows my config does work but not in you app. When i try to open the encrypted folder configuration it just throws me the error message "Error retrieving directory content".

If i use the Team Drive configuration only, without encryption, i can see files and folders. The Cache configuration does not work too.

How can i debug this issue or provide you a log file (could not find it). Thanks a lot again!

Or can you share a working config so that i can edit mine? Not sure how the cache config should look like, etc. 

THANKS

**Comment by kaczmarkiewiczp** [comment]
Thanks for your support.

In order to determine where the problem with TeamDrive is I will need your log file. If you haven't done so already, please enable file logging under settings (Use logs). Next, try to open your encrypted TeamDrive in order to get the "Error retrieving directory content" message. The full error should now be saved to the log file. The log file is located at Android/data/ca.pkay.rcloneexplorer/files/logs/log.txt
Please paste the log file or its content here, and I'll try to fix the problem.

Now about cache remote, unfortunately I don't think that can be supported in the app. Seems to be a problem with how rclone is compiled for Android, and rclone cache needing to create temporary files in places it doesn't have access. I was only able to fix it to the point where cache remote can be viewed, but any operation (download, upload, move, delete, etc.) fails. This would make it pretty much useless. I will continue working on this, but can't promise any timeline or if the issue can be fixed at all. In the meantime I might hide cache remotes from the app and show a prompt that they are not supported.

**Comment by jpdribbler** [comment]
Thanks a lot for your fast reply. I will never ask for ETA! :)

Here is my log file. Using a Google Pixel XL
```

2018-05-14 19:56:05 - 2018/05/14 17:56:05 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 19:56:08 - 2018/05/14 17:56:08 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 19:56:10 - 2018/05/14 17:56:10 Failed to create file system for "GSuiteCache:": failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 19:56:47 - 2018/05/14 17:56:47 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 19:57:53 - 2018/05/14 17:57:53 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 19:58:45 - 2018/05/14 17:58:45 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 19:59:29 - 2018/05/14 17:59:29 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:06:43 - 2018/05/14 20:06:43 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:06:44 - 2018/05/14 20:06:44 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:06:46 - 2018/05/14 20:06:46 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:06:52 - 2018/05/14 20:06:52 Failed to create file system for "GSuiteCache:": failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:06:53 - 2018/05/14 20:06:53 Failed to create file system for "GSuiteCache:": failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:07:07 - 2018/05/14 20:07:07 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:07:08 - 2018/05/14 20:07:08 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:07:44 - 2018/05/14 20:07:44 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:11:18 - 2018/05/14 20:11:18 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:11:22 - 2018/05/14 20:11:22 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-14 22:11:27 - 2018/05/14 20:11:27 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-21 07:31:33 - 2018/05/21 05:31:33 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-21 07:36:40 - 2018/05/21 05:36:40 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-21 07:37:38 - 2018/05/21 05:37:38 Failed to create file system for "GSuiteCache:": failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-29 19:45:31 - 2018/05/29 17:45:31 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-29 19:45:52 - 2018/05/29 17:45:52 Failed to create file system for "GSuiteCache:": failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
2018-05-29 19:45:55 - 2018/05/29 17:45:55 Failed to create file system for "Gsuitecrypt:": failed to make remote "GSuiteCache:" to wrap: failed to create cache directory /data/data/com.termux/files/usr/tmp/rclone/cache-backend: mkdir /data/data/com.termux: permission denied
```

Also my configuration file which i use:

```
[GsuiteTeam]
type = drive
client_id = secret
client_secret = secret
scope = drive
root_folder_id = 
service_account_file = 
token = {"access_token":"secret"}
team_drive = secret

[Gsuitecrypt]
type = crypt
remote = GSuiteCache:
filename_encryption = standard
directory_name_encryption = true
password = secret
password2 = secret

[GSuiteCache]
type = cache
remote = GsuiteTeam:
plex_url = http://127.0.0.1:32400
plex_username = secret
plex_password = secret
chunk_size = 10M
info_age =
chunk_total_size =


```

**Comment by kaczmarkiewiczp** [comment]
The underlying problem is the GSuiteCache remote. Once I fix the problem with cache remotes not working in the app, you should be able to access your encrypted TeamDrive. I should have a fix ready in a day or two max.
In the meantime, you could create an encrypted remote that points to your TeamDrive (skipping the cache). You'd have to do it from your computer, as creating TeamDrives from the app won't be supported until release of rclone v1.42 (due to some bugs in rclone).

**Comment by jpdribbler** [comment]
Nice. That was easy. After you said that the cache is not supported i should have done this by myself... 👍 

It works! Perfect, thanks a lot!

Is there any benefit in performance if you enable the cache?

**Comment by jpdribbler** [comment]
One more question:

Unfortunately i am not able to open one folder in Google Drive teamdrive with more than 400 files and folders (~4TB of data). Is there something i can do to fix this? Thanks again!

**Comment by kaczmarkiewiczp** [comment]
Cache remotes should now work correctly in v1.3.2.
I'm not certain how exactly rclone cache works, but I believe that there should be a performance improvement, especially on slower networks. The big advantage I've noticed is that once the file structure is cached, it's possible to view the file structure even when there's no internet connection.

Now about the issue with viewing folders with large amount of files. I was able to replicate it on my end. It seems to be caused by large amount of files in a folder, and not file/folder size. I'll try to determine what exactly is the problem and fix it.

Thanks for using my app, and for reporting :smiley:
